javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * closely translated from:
+ * https://github.com/scala/scala
+ * scala.concurrent.fokjoin.ThreadLocalRandom
+ */
+
+package java.util.concurrent
+
+import java.util.Random
+
+class ThreadLocalRandom extends Random {
+
+  override protected def _setSeed(seed_in: Long): Unit = ()
+
+  override def setSeed(seed: Long): Unit =
+    throw new UnsupportedOperationException()
+
+  /*
+  * this code has to be removed!!!
+  */
+  import scala.scalajs.js
+  /** Generate a random long from JS RNG to seed a new Random */
+  private def randomSeed(): Long = {
+    val h = randomInt().toLong
+    val l = randomInt().toLong
+    (h << 32) | (l & 0xffffffffL)
+  }
+
+  private def randomInt(): Int =
+    (Math.floor(js.Math.random() * 4294967296.0) - 2147483648.0).toInt
+  /*
+  * end of code to be removed
+  */
+
+
+  def nextInt(least: Int, bound: Int): Int = {
+    if (least >= bound) throw new IllegalArgumentException()
+
+    //nextInt(bound - least) + least
+    (nextLong(bound.toLong - least.toLong) + least.toLong).toInt
+  }
+
+  def nextLong(_n: Long): Long = {
+    if (_n <= 0)
+      throw new IllegalArgumentException("n must be positive")
+    // Divide n by two until small enough for nextInt. On each
+    // iteration (at most 31 of them but usually much less),
+    // randomly choose both whether to include high bit in result
+    // (offset) and whether to continue with the lower vs upper
+    // half (which makes a difference only if odd).
+    var offset: Long = 0L
+    var n: Long = _n
+
+    while (n >= Integer.MAX_VALUE) {
+      val bits: Int = next(2)
+      val half: Long = n >>> 1
+      val nextn =
+        if ((bits & 2) == 0) half
+        else n - half
+      if ((bits & 1) == 0)
+        offset += n - nextn
+      n = nextn
+    }
+    offset + nextInt(n.toInt)
+  }
+
+  def nextLong(least: Long, bound: Long): Long = {
+    if (least >= bound)
+      throw new IllegalArgumentException()
+
+    (nextDouble(bound.toDouble - least.toDouble) + least.toDouble).toLong
```

Comment:
Then you need to figure something out, because your solution does not work in corner cases either. Also when you are close to the upper bound of the type.